### DISCUSSION TITLE
Hackathon Topics

### DISCUSSION THREAD

**Discussion by mjohnson541** [ReactionMechanismGenerator/RMG-Py#2087] [Q&A]
We haven't had a hackathon in a while and we're starting to build up issues we'd normally resolve there. In particular running regression tests on Github Actions and general switching from travis to github actions. Please add any ideas that occur to you and we can put together a list of hackathon topics (hopefully some for more advanced RMG developers and some for less experienced developers).

**Comment by rwest** [comment]
Some ideas to get started. Maybe we should have a [project](https://github.com/ReactionMechanismGenerator/RMG-Py/projects) then we can reorder them and track them.

1. updating the RMG fork of cclib to use OpenBabel 3, and making new conda builds for both Linux and Mac https://github.com/cclib/cclib/commit/3f73c94d1cb5c62fe5ba15af46e204fd2abab9da
2. fixing other cclib problems like https://github.com/ReactionMechanismGenerator/RMG-Py/issues/1864 https://github.com/ReactionMechanismGenerator/RMG-Py/issues/1875
2. switching away from using an RMG fork of cclib and using the official ccilb, potentially contributing our improvements upstream https://github.com/ReactionMechanismGenerator/RMG-Py/issues/756
3. building new conda binaries of all the RMG-maintained conda packages, for a matrix of python versions and OS. automating this. (eg. https://github.com/ReactionMechanismGenerator/RMG-Py/issues/2047)
4. completing the switch over to OpenBabel 3 https://github.com/ReactionMechanismGenerator/RMG-Py/pull/2088
5. github actions for documentation builds https://github.com/ReactionMechanismGenerator/RMG-Py/pull/2073
6. ~github actions for unit tests https://github.com/ReactionMechanismGenerator/RMG-Py/pull/2072~
7. github actions for regression tests https://github.com/ReactionMechanismGenerator/RMG-tests
8. github actions to test all example input files don't crash https://github.com/ReactionMechanismGenerator/RMG-Py/issues/2040
8. add catalysis example to regression tests https://github.com/ReactionMechanismGenerator/RMG-tests/pull/71
9. other improvements to regression tests https://github.com/ReactionMechanismGenerator/RMG-tests/issues
10. 3.1 release https://github.com/ReactionMechanismGenerator/RMG-Py/issues/2047
11.  make the yaml output in cantera format ~not~ *as well as* rms format (rms plans to switch to cantera format anyway) https://github.com/ReactionMechanismGenerator/RMG-Py/issues/2078
12. Cleaning up the issue tracker and closing some old and irrelevant issues. 
13. rmg-tests haven't been running on RMG-database PRs for over a month. [Example](https://travis-ci.org/github/ReactionMechanismGenerator/RMG-tests/builds/758183430)

**Comment by rwest** [comment]
Looking at the week of April 12th. Watch out for a whentomeet poll.

**Reply by rwest** [reply]
https://www.when2meet.com/?11513288-j4ZmH

**Comment by rwest** [comment]
We should go through the issues tracker and flag some, or put them in a "project"

**Reply by rwest** [reply]
Here's a project. please add cards . https://github.com/ReactionMechanismGenerator/RMG-Py/projects/7

**Comment by rwest** [comment]
For people watching this discussion topic for news about the hackathon. It's tomorrow (Thursday) 9am-5pm Eastern time. Message someone if you didn't get the zoom link in your email.
